R/quanteda_options.R

```diff
@@ -1,5 +1,5 @@
 # implement default options
-QUANTEDA_OPTIONS <- list(threads = max(1L, floor(RcppParallel::defaultNumThreads() / 2)),
+QUANTEDA_OPTIONS <- list(threads = max(1L, RcppParallel::defaultNumThreads() - 1),
```

Comment:
We previously had n-1 cores, where n = all available, but I change it to 1/2 for a reason that now escapes me. Can we think of any reason not to take all but one of a user's cores?

We also want to test that if this is set in `.Rprofile` then it stays set. (This is not possible with **testthat** and therefore needs to be done manually.)

Reply:
I set n - 1, because sustained 100% CPU load can make system unstable - sometimes even not able to open task manager. There is a test that directly compare the option value and the environment variable in the test:
```r
expect_equal(
        quanteda_options("threads"),
        as.numeric(Sys.getenv('RCPP_PARALLEL_NUM_THREADS'))
    )
```